data/Builder.go

```diff
@@ -0,0 +1,72 @@
+package data
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/tidepool-org/platform/validate"
+)
+
+type Datum map[string]interface{}
+
+type DatumArray []Datum
+
+type Builder interface {
+	BuildFromDatum(datum Datum) interface{}
+	BuildFromDatumArray(datumArray DatumArray) ([]interface{}, *validate.ErrorsArray)
+}
+
+type TypeBuilder struct {
+	inject map[string]interface{}
+	Errors *validate.ErrorsArray
+	Index  int
+}
+
+func NewTypeBuilder(inject map[string]interface{}) Builder {
+	return &TypeBuilder{
+		inject: inject,
+		Errors: validate.NewErrorsArray(),
+		Index:  0,
+	}
+}
+
+func (t *TypeBuilder) BuildFromDatumArray(datumArray DatumArray) ([]interface{}, *validate.ErrorsArray) {
+
+	var set []interface{}
+
+	for i := range datumArray {
+		if item := t.BuildFromDatum(datumArray[i]); item != nil {
```

Comment:
What happens if `item` is nil? Can this ever happen? If so, does it mean that there is one or more errors? If so, we're just going to return those errors to the user and the resultsArray will be ignored.  So, I see no reason not to just simplify this down to:

```
resultsArray = append(resultsArray, t.BuildFromDatum(datumArray[i]))
```

What do you think?